src/harness/reference_models/ppa/ppa.py

```diff
@@ -0,0 +1,150 @@
+#    Copyright 2016-2018 SAS Project Authors. All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+
+import multiprocessing
+
+import geojson
+import numpy as np
+from concurrent import futures
+from reference_models.antenna import antenna
+from reference_models.geo import census_tract, vincenty, nlcd
+from reference_models.geo import utils
+from reference_models.propagation import wf_hybrid
+from shapely import geometry, ops
+
+THRESHOLD_PER_10MHZ = -96
+RX_HEIGHT = 1.5
+MAX_ALLOWABLE_EIRP_PER_10_MHZ_CAT_A = 30.
+MAX_ALLOWABLE_EIRP_PER_10_MHZ_CAT_B = 47.
+census_tract_driver = census_tract.CensusTractDriver()
+nlcd_driver = nlcd.NlcdDriver()
+
+
+def _CalculateDbLossForEachPointAndGetContour(install_param, antenna_gain, latitudes, longitudes):
+  db_loss = np.zeros(len(latitudes), dtype=np.float64)
+  for index, lat_lon in enumerate(zip(latitudes, longitudes)):
+    lat, lon = lat_lon
+    code = nlcd_driver.GetLandCoverCodes(lat, lon)
+    region_type = nlcd.GetRegionType(code)
+    db_loss[index] = wf_hybrid.CalcHybridPropagationLoss(install_param['latitude'],
+                                                         install_param['longitude'],
+                                                         install_param['height'],
+                                                         lat, lon, RX_HEIGHT,
+                                                         install_param['indoorDeployment'],
+                                                         reliability=0.5,
+                                                         region=region_type).db_loss
+  index_cond, = np.where(install_param['eirpCapability'] + antenna_gain - db_loss > THRESHOLD_PER_10MHZ)
+  return index_cond.shape[0] / 5.
+
+
+def _HammingFilter(x, window_len=15):
+  s = np.r_[x[-(window_len / 2):], x, x[0:window_len / 2]]
+  w = np.hamming(window_len)
+  y = np.convolve(w / w.sum(), s, mode='valid')
+  return y
+
+
+def _GetPolygon(device):
```

Comment:
Small docstring would be useful, for example:
"""Returns the PPA contour for a single CBSD device, as a shapely polygon."""